packages/nocodb/src/services/hooks.service.ts

```diff
@@ -46,6 +65,18 @@ export class HooksService {
     validatePayload('swagger.json#/components/schemas/HookReq', param.hook);
 
     this.validateHookPayload(param.hook.notification);
+    const path = (param.hook.notification as any).payload.path || '';
+    const webhookUrl = this.replaceEnvVariable(path);
+
+    if (path) {
+      param.hook.notification = {
+        ...(param.hook.notification as Record<string, any>),
+        payload: {
+          ...((param.hook.notification as Record<string, any>).payload || {}),
+          path: webhookUrl,
+        },
+      };
+    }
```

Comment:
Make this with the one below to be one function called from both functions since they do the same thing